docs/docs/guides/web3_config/index.md

```diff
@@ -201,6 +202,610 @@ console.log(web3.getContextObject().config)
+
+console.log(web3.getContextObject().config);
+```
+:::info
+The `defaultNetworkId` can be configured both globally and at the package level:
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.eth.defaultNetworkId = undefined;
+
+console.log(web3.eth.getContextObject().config);
+```
+:::
+
+### defaultChain
+The `defaultChain` is used for building the `baseChain` property of the tx options in a transaction. If the `defaultCommon.basechain` is set, the`defaultChain` should be consistent with it.
+
+The default value of `defaultChain` is `mainnet`.
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.defaultChain = 'mainnet';
+
+console.log(web3.getContextObject().config);
+```
+:::info
+The `defaultChain` can be configured both globally and at the package level:
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.eth.defaultChain = 'mainnet';
+
+console.log(web3.eth.getContextObject().config);
+```
+:::
+#### All available choices for defaultChain:
+```ts
+'goerli' | 'kovan' | 'mainnet' | 'rinkeby' | 'ropsten' | 'sepolia'
+```
+
+
+### defaultHardfork
+The `defaultHardfork` is used for building the `defaultHardfork` property of the tx options in a transaction. If the `defaultCommon.defaultHardfork` is set, the`defaultHardfork` should be consistent with it.
+
+The default value of `defaultHardfork` is `london`.
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.defaultHardfork = 'london';
+
+console.log(web3.getContextObject().config);
+```
+:::info
+The `defaultHardfork` can be configured both globally and at the package level:
+```ts
+import { Web3 } from 'web3';
+
+const web3 = new Web3('http://127.0.0.1:7545');
+
+web3.eth.defaultHardfork = 'london';
+
+console.log(web3.eth.getContextObject().config);
+```
+:::
+#### All available choices for contractDataInputFill:
+```ts
+'chainstart'
+'frontier'
+'homestead'
+'dao'
+'tangerineWhistle'
+'spuriousDragon'
+'byzantium'
+'constantinople'
+'petersburg'
+'istanbul'
+'muirGlacier'
+'berlin'
+'london'
+'altair'
+'arrowGlacier'
+'grayGlacier'
+'bellatrix'
+'merge'
+'capella'
+'sharding'
+```
+
+### defaultCommon
+The `defaultCommon` is used for building the `common` property of the tx options in a transaction. It should be consistent with the `defaultHardfork` and `defaultChain`. The `defaultCommon` property does contain the following Common object:
+```
+- customChain - Object: The custom chain properties
+  - name - string: (optional) The name of the chain
+  - networkId - number: Network ID of the custom chain
+  - chainId - number: Chain ID of the custom chain
+- baseChain - string: (optional) 'goerli', 'kovan', 'mainnet', 'rinkeby', 'ropsten' or 'sepolia'
+- hardfork - string: (optional) chainstart, homestead, dao, tangerineWhistle, spuriousDragon, byzantium, constantinople, petersburg, istanbul, berlin, or london
+```
```

Comment:
It may be better to just link to [the API docs](https://docs.web3js.org/libdocs/Contract#defaultcommon) for the `defaultCommon` property.